# Importing Libraries

In [2]:
# importing libraries
import pandas as pd
import numpy as np
import os

# Importing data

In [4]:
df= pd.read_csv(r'C:\users\Ali Hassan\Documents\MM-2020 Instacart Basket Analysis\Data\Original data\orders.csv', index_col =False)

In [5]:
# Define the project folder path
path= r'C:\users\Ali Hassan\Documents\MM-2020 Instacart Basket Analysis'

In [6]:
path

'C:\\users\\Ali Hassan\\Documents\\MM-2020 Instacart Basket Analysis'

In [38]:
#Define the path to data file
df= pd.read_csv(os.path.join(path, 'data','original data','orders.csv'), nrows=1000)

In [40]:
#Define the columns to include (excluding 'eval_set')
vars_list= [ 'order_id','user_id','order_number','order_dow','order_hour_of_day','days_since_prior_order']

In [42]:
vars_list

['order_id',
 'user_id',
 'order_number',
 'order_dow',
 'order_hour_of_day',
 'days_since_prior_order']

In [44]:
df= pd.read_csv(os.path.join(path,'data','original data','orders.csv'), usecols = vars_list)

In [49]:
#Display the first five rows of 'orders.csv' dataset
df.head(5)

order_id  user_id  order_number  order_dow  order_hour_of_day  \
0   2539329        1             1          2                  8   
1   2398795        1             2          3                  7   
2    473747        1             3          3                 12   
3   2254736        1             4          4                  7   
4    431534        1             5          4                 15   

   days_since_prior_order  
0                     NaN  
1                    15.0  
2                    21.0  
3                    29.0  
4                    28.0

In [53]:
#Define the path to 'product_csv' file
df_prods= pd.read_csv(os.path.join(path, 'data','original data','products.csv'), nrows=1000)

In [55]:
#Display the firts 20 rows of 'products_csv' data set.
df_prods.head(20)

product_id                                       product_name  aisle_id  \
0            1                         Chocolate Sandwich Cookies        61   
1            2                                   All-Seasons Salt       104   
2            3               Robust Golden Unsweetened Oolong Tea        94   
3            4  Smart Ones Classic Favorites Mini Rigatoni Wit...        38   
4            5                          Green Chile Anytime Sauce         5   
5            6                                       Dry Nose Oil        11   
6            7                     Pure Coconut Water With Orange        98   
7            8                  Cut Russet Potatoes Steam N' Mash       116   
8            9                  Light Strawberry Blueberry Yogurt       120   
9           10     Sparkling Orange Juice & Prickly Pear Beverage       115   
10          11                                  Peach Mango Juice        31   
11          12                         Chocolate Fudge Layer Cake       119   
12          13                                  Saline Nasal Mist        11   
13          14                     Fresh Scent Dishwasher Cleaner        74   
14          15                           Overnight Diapers Size 6        56   
15          16                      Mint Chocolate Flavored Syrup       103   
16          17                                  Rendered Duck Fat        35   
17          18                Pizza for One Suprema  Frozen Pizza        79   
18          19   Gluten Free Quinoa Three Cheese & Mushroom Blend        63   
19          20     Pomegranate Cranberry & Aloe Vera Enrich Drink        98   

    department_id  prices  
0              19     5.8  
1              13     9.3  
2               7     4.5  
3               1    10.5  
4              13     4.3  
5              11     2.6  
6               7     4.4  
7               1     1.1  
8              16     7.0  
9               7     8.4  
10              7     2.8  
11              1     9.4  
12             11     1.1  
13             17     6.5  
14             18    11.2  
15             19     5.2  
16             12    17.1  
17              1    10.6  
18              9    12.0  
19              7     6.0

In [57]:
#Display the last 35 rows of 'products_csv' dataset.
df_prods.tail(35)

product_id                                      product_name  aisle_id  \
965         965             Maxtra Cartridge Water Filter Refills        87   
966         966                                   Pear Hard Cider        27   
967         967   Gluten Free Soft & Chewy Snicker Doodle Cookies         8   
968         968                      Zipper Bags Sandwich - 50 CT        85   
969         969                                 Petite Green Peas       116   
970         970              Salmon and Tuna Formula Dry Cat Food        41   
971         971                       Chicken Alfredo with Rotini        38   
972         972                               Linguine No 7 Pasta       131   
973         973                               8 X 6 Cutting Board        10   
974         974              Organic 54% Cocoa Dark Chocolate Bar        45   
975         975            Creamy Vanilla Soy Milk Frozen Dessert        37   
976         976       Cavity Protection Regular Flavor Toothpaste        20   
977         977                          Deli Spicy Brown Mustard        72   
978         978                               Mallow Oats Cereals       121   
979         979  Odor Control Home Trash Compactor Bags 18 Gallon        60   
980         980         Automatic Toilet Bowl Cleaner with Bleach       114   
981         981                                       Garlic Naan       128   
982         982                                 Aged Gouda Cheese         2   
983         983                            Strawberry Pomegranate         3   
984         984               Cucumber Witch Hazel With Aloe Vera       109   
985         985                                Peanut Butter Bars         3   
986         986                   Crushed Pineapple in 100% Juice        99   
987         987   Naturals Antiseptic Adult Mouthwash Herbal Mint        20   
988         988       Cranberrry  Almond Soft & Chewy Granola Bar         3   
989         989                Chapatti Chicken Tikki Masala Wrap        38   
990         990                                    Diced Pancetta        96   
991         991                             Red Sweet Mini Pepper       100   
992         992                Thin Crust Pepperoni & Pesto Pizza        79   
993         993             Mild Diced Tomatoes with Green Chilis        81   
994         994               Shake n Pour Buttermilk Pancake Mix       130   
995         995                 Sausage Smoked Sweet German Brand       106   
996         996                Honey Cinnamon Nut-Thins  Crackers       125   
997         997              Mini Double Chocolate Ice Cream Bars        37   
998         998                           Hot Chopped Green Chili       104   
999         999                  Original Organic Ville BBQ Sauce         5   

     department_id  prices  
965             17     1.4  
966              5     2.6  
967              3     2.4  
968             17     4.6  
969              1     4.8  
970              8     9.9  
971              1     7.8  
972              9     4.5  
973             17     5.0  
974             19     4.4  
975              1     3.9  
976             11    14.6  
977             13    13.7  
978             14     4.4  
979             17     5.7  
980             17     8.7  
981              3     3.8  
982             16    14.1  
983             19     3.0  
984             11     8.4  
985             19     6.4  
986             15    13.1  
987             11     9.4  
988             19     6.3  
989              1    12.6  
990             20    13.6  
991             21     5.4  
992              1    10.2  
993             15    10.2  
994             14    13.4  
995             12    16.7  
996             19     3.2  
997              1     8.5  
998             13     9.1  
999             13     9.2

In [59]:
#Display the column names in 'product_csv' dataset.
df_prods.columns

Index(['product_id', 'product_name', 'aisle_id', 'department_id', 'prices'], dtype='object')

In [61]:
#Display the number of rows and column in "product_csv' dataset.
df_prods.shape

(1000, 5)

In [69]:
# Find the maximum value of the 'aisle_id' in df_pandas.
max_aisle_id= df_prods['aisle_id']. max()

In [71]:
print("Maximum value of aisle_id:",max_aisle_id)

Maximum value of aisle_id: 134


In [73]:
# Check the data type of the 'department_id' in df_prods.
dept_id_type= df_prods['department_id'].dtype

In [75]:
print("data type of deppartment_id:",dept_id_type)

data type of deppartment_id: int64


# Answers
1. The maximum value of 'aisle_id' in df_pandas is 134.
2. The data type of 'department_id' in df_prods is 'int64'.